generator_process/actions/huggingface_hub.py

```diff
@@ -0,0 +1,82 @@
+from dataclasses import dataclass
+import os
+
+@dataclass
+class Model:
+    id: str
+    author: str
+    tags: list[str]
+    likes: int
+    downloads: int
+
+def hf_list_models(
+    self,
+    query: str
+) -> list[Model]:
+    from huggingface_hub import HfApi, ModelFilter
+    
+    if hasattr(self, "huggingface_hub_api"):
+        api: HfApi = self.huggingface_hub_api
+    else:
+        api = HfApi()
+        setattr(self, "huggingface_hub_api", api)
+    
+    filter = ModelFilter(tags="diffusers", task="text-to-image")
+    models = api.list_models(
+        filter=filter,
+        search=query
+    )
+
+    return list(map(lambda m: Model(m.modelId, m.author, m.tags, m.likes, getattr(m, "downloads", -1)), models))
+
+def hf_list_installed_models(self) -> list[Model]:
+    from diffusers.utils import DIFFUSERS_CACHE
+    if not os.path.exists(DIFFUSERS_CACHE):
+        return []
+    return list(
+        filter(
+            lambda x: os.path.isdir(x.id),
+            map(lambda x: Model(os.path.join(DIFFUSERS_CACHE, x), "", [], -1, -1), os.listdir(DIFFUSERS_CACHE))
+        )
+    )
+
+def hf_snapshot_download(
+    self,
+    model: str,
+    token: str
+) -> None:
+    from huggingface_hub import snapshot_download
+    from diffusers import StableDiffusionPipeline
+    from diffusers.utils import DIFFUSERS_CACHE, WEIGHTS_NAME, CONFIG_NAME, ONNX_WEIGHTS_NAME
+    from diffusers.schedulers.scheduling_utils import SCHEDULER_CONFIG_NAME
+    from diffusers.hub_utils import http_user_agent
+    config_dict = StableDiffusionPipeline.get_config_dict(
+        model,
+        cache_dir=DIFFUSERS_CACHE,
+        resume_download=True,
+        force_download=False,
+        use_auth_token=token
+    )
+    # make sure we only download sub-folders and `diffusers` filenames
+    folder_names = [k for k in config_dict.keys() if not k.startswith("_")]
+    allow_patterns = [os.path.join(k, "*") for k in folder_names]
+    allow_patterns += [WEIGHTS_NAME, SCHEDULER_CONFIG_NAME, CONFIG_NAME, ONNX_WEIGHTS_NAME, StableDiffusionPipeline.config_name]
+
+    # make sure we don't download flax weights
+    ignore_patterns = "*.msgpack"
```

Comment:
Just a heads-up :-) Soon also safe tensors might be added to many diffusion repos since they are safe (compared to pickle) and fast. E.g. see: https://huggingface.co/CompVis/stable-diffusion-v1-4/discussions/171

So this might have to be added there if needed :-) 

The use case here is that you would like to download the model, but not load it into memory no? 
Think that's more or less exactly what @keturn was asking for as well

Maybe, it's time to better modularize `from_pretrained(...)` in `diffusers` to make this part of the code more robust :-)

Reply:
Yes, exactly. Something like `hf_hub_download` for `diffusers` models would be great!